ui/components/multichain/pages/send/components/recipient.tsx

```diff
@@ -10,93 +10,98 @@ import {
 } from '../../../../../ducks/send';
 import {
   getDomainError,
-  getDomainResolution,
+  getDomainResolutions,
+  getDomainType,
   getDomainWarning,
 } from '../../../../../ducks/domains';
 import {
   BannerAlert,
   BannerAlertSeverity,
   Box,
 } from '../../../../component-library';
-import { getAddressBookEntry } from '../../../../../selectors';
 import { Tab, Tabs } from '../../../../ui/tabs';
-import { AddressListItem } from '../../../address-list-item';
+import { DomainInputResolutionCell } from './domain-input-resolution-cell';
 import { SendPageAddressBook, SendPageRow, SendPageYourAccounts } from '.';
 
 const CONTACTS_TAB_KEY = 'contacts';
 const ACCOUNTS_TAB_KEY = 'accounts';
 
-const ENS_RESOLUTION_TYPE = 'ENS resolution';
-
-const renderExplicitAddress = (
-  address: string,
-  nickname: string,
-  type: string,
-  // TODO: Replace `any` with type
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  dispatch: any,
-) => {
-  return (
-    <AddressListItem
-      address={address}
-      label={nickname}
-      useConfusable={type === ENS_RESOLUTION_TYPE}
-      onClick={() => {
-        dispatch(
-          addHistoryEntry(
-            `sendFlow - User clicked recipient from ${type}. address: ${address}, nickname ${nickname}`,
-          ),
-        );
-        dispatch(updateRecipient({ address, nickname }));
-        dispatch(updateRecipientUserInput(address));
-      }}
-    />
-  );
-};
-
 export const SendPageRecipient = () => {
   const t = useContext(I18nContext);
   const dispatch = useDispatch();
 
   const recipient = useSelector(getRecipient);
   const userInput = useSelector(getRecipientUserInput) || '';
 
-  const domainResolution = useSelector(getDomainResolution);
+  const domainResolutions = useSelector(getDomainResolutions) || [];
```

Comment:
```suggestion
  const domainResolutions = useSelector(getDomainResolutions) ?? [];
```

Reply:
But why is this needed in the first place? 🤔

Reply:
initial state for domain resolutions is `null`.